package/windows/launchdf.c

```diff
@@ -12,28 +31,40 @@ int WINAPI wWinMain(HINSTANCE hi, HINSTANCE hpi, PWSTR cmd, int ns)
     WCHAR steamPath[1024];
     DWORD datasize = 1024;
 
-    LONG retCode = RegGetValueW(HKEY_CURRENT_USER, L"SOFTWARE\\Valve\\Steam", L"SteamExe", RRF_RT_REG_SZ, NULL, &steamPath, &datasize);
+    LONG retCode = RegGetValueW(HKEY_CURRENT_USER, L"SOFTWARE\\Valve\\Steam",
+            L"SteamExe", RRF_RT_REG_SZ, NULL, &steamPath, &datasize);
 
     if (retCode != ERROR_SUCCESS)
-    {
-        MessageBoxW(NULL, L"Could not find Steam client executable", NULL, 0);
-        exit(1);
-    }
+        return L"Could not find Steam client executable";
 
     WCHAR commandLine[1024] = L"steam.exe -applaunch 975370";
 
-    if (CreateProcessW(steamPath,
-        commandLine,
-        NULL,
-        NULL,
-        FALSE,
-        0,
-        NULL,
-        NULL,
-        &si,
-        &pi) == 0)
-    {
-        MessageBoxW(NULL, L"could not launch Dwarf Fortress", NULL, 0);
+    if (CreateProcessW(steamPath, commandLine,
+            NULL, NULL, FALSE, 0, NULL, NULL, &si, &pi) == 0)
+        return L"Could not launch Dwarf Fortress";
+
+    return NULL;
+}
+
+// this method doesn't properly attribute Steam playtime metrics to DF,
+// but that's better than not having DF start at all.
+static BOOL launch_direct() {
+    STARTUPINFOA si;
+    PROCESS_INFORMATION pi;
+
+    ZeroMemory(&si, sizeof(si));
+    si.cb = sizeof(si);
+    ZeroMemory(&pi, sizeof(pi));
+
+    return CreateProcessA("Dwarf Fortress.exe",
```

Comment:
we used `CreateProcessW` earlier but `CreateProcessA` here. this is technically not really a problem since `"Dwarf Fortress.exe"` works fine as a narrow string in this context, but the nonparallelism irks me a bit

for consistency i'd probably use `CreateProcessW(L"Dwarf Fortress.exe",` here. you'll also have to replace `STARTUPINFOA si` with `STARTUPINFOW si` above.

this might become more important if we ever need to actually keep the launcher running for some reason (workshop interaction, perhaps), in which case we might actually have to use the `si` and `pi` values later on, which means they need to be the same type in either launch scenario, and using the unicode versions elsewhere provides for better i18n support

Reply:
done